micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/HazelcastCacheMetrics.java

```diff
@@ -138,23 +153,23 @@ protected void bindImplementationSpecificMetrics(MeterRegistry registry) {
     }
 
     private void nearCacheMetrics(MeterRegistry registry) {
-        if (cache.getLocalMapStats().getNearCacheStats() != null) {
-            Gauge.builder("cache.near.requests", cache, cache -> cache.getLocalMapStats().getNearCacheStats().getHits())
+        if (cache.getLocalMapStats() != null && cache.getLocalMapStats().getNearCacheStats() != null) {
+            Gauge.builder("cache.near.requests", cache, cache -> cache.getLocalMapStats() != null ? cache.getLocalMapStats().getNearCacheStats().getHits() : Double.NaN)
```

Comment:
At this point, `cache.getLocalMapStats()` can't be null, right?

Reply:
Hmm, good question. I think you may be right, but it's a bit confusing with the subtle dependencies between the two classes now (`HazelcastCacheMetrics` and `HazelcastIMapAdapter`).

The method is annotated with `@Nullable` though, so removing the `null` check will cause IntelliJ to complain. :unamused: But I can definitely remove the explicit `null` checks if you want me to.

Reply:
I dug deeper on this, I don't think anymore that my previous comment is valid. :)
This seemed a little weird for first sight but it makes perfect sense for the second, thanks.